tecken/download/views.py

```diff
@@ -80,7 +107,27 @@ def download_symbol(request, symbol, debugid, filename):
             code_id=request.GET.get('code_id'),
         )
 
-    response = http.HttpResponseNotFound('Symbol Not Found')
+        if (
+            symbol.lower().endswith('.pdb') and
+            filename.lower().endswith('.sym')
+        ):
+            # If we haven't already sent it to the 'download_microsoft_symbol'
+            # background task, do so.
+
+            download_from_microsoft(symbol, debugid)
+
+            downloader.invalidate_cache(symbol, debugid, filename)
+
+            # The querying of Microsoft's server is potentially slow.
+            # That's why this call is down in a celery task.
+            # But there is hope! And the client ought to be informed
+            # that if they just try again in a couple of seconds/minutes
+            # it might just be there.
+            delayed_lookup = True
+
+    response = http.HttpResponseNotFound(
+        'Symbol Not Found Yet' if delayed_lookup else 'Symbol Not Found'
```

Comment:
What do you think about this crazy idea (thanks @lonnen for inspiring this idea).

Instead of responding with `404` we can instead respond with `302` (or some other 3xx) code. But when we set the `Location` header we add a little extra to the URL like `&was-recently-put-in-background-task=1` and when the client comes back to us again, this time we put in a little sleep so that `curl` doesn't fail on an "infinite" redirect loop **too quickly**. 

It would be nice because then things like stackwalker wouldn't need to do anything smarter to retry based on the output.